src/tensors/cpu/prod_blas.h

```diff
@@ -1,11 +1,117 @@
+#pragma once
 #if MKL_FOUND
-#include <mkl.h>
-#else
-#if BLAS_FOUND
-#include <cblas.h>
-#endif
+    #include <mkl.h>
+#elif BLAS_FOUND
+    #include <cblas.h>
+#elif USE_RUY_SGEMM
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcomment"
+    #include "ruy/ruy.h"
+    #include "ruy/system_aligned_alloc.h"
+#pragma GCC pop
 #endif
 
+#if USE_RUY_SGEMM
```

Comment:
I am not particularly happy about this.

Reply:
Rome wasn't built in a day

Reply:
Do you have an M1 to test on ? We should test RUY vs OpenBLas vs AppleAccelerate to determine what sgemm implementation to use here.

Reply:
On the case with M1

Reply:
Ruy path was intended for Android. The usual BLAS was [painful to compile](https://github.com/OpenMathLib/OpenBLAS/wiki/How-to-build-OpenBLAS-for-Android). Since not available via package manager or anything we'd have to go compile-route with a submodule addition if going for OpenBLAS.

Reply:
Did you guys ever get the int8 stuff to work? Did you use RUY?

Reply:
I haven't ported it from our fork yet. Int8 with ruy works in our port.

Reply:
ah right. Where's your code? We'll take a look. May give you a hand with it

Reply:
https://github.com/browsermt/marian-dev/blob/master/src/tensors/cpu/ruy_interface.h